rq/queue.py

```diff
@@ -106,6 +106,9 @@ def is_empty(self):
         return self.count == 0
 
     def fetch_job(self, job_id):
+        if job_id not in self.job_ids:
```

Comment:
fixed. I think I tried this to start with by couldn't find the reference to the queue form the job because it's rather confusingly called `.origin`.